pkg/operation/botanist/etcdencryption.go

```diff
@@ -0,0 +1,332 @@
+	secretName := fmt.Sprintf("%s.%s", b.Shoot.Info.Name, common.EtcdEncryptionSecretName)
+	return secretName
+}
+
+// readEncryptionConfigurationFromGarden reads the EncryptionConfiguration from the shoot namespace in the seed
+func (b *Botanist) readEncryptionConfigurationFromGarden() (bool, *apiserverconfigv1.EncryptionConfiguration, error) {
+	client := b.Operation.K8sGardenClient
+	secretNameInGardenCluster := b.calculateEtcdEncryptionSecretNameInGardenCluster()
+	ecs, err := client.GetSecret(b.Shoot.Info.Namespace, secretNameInGardenCluster)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil, nil
+		} else {
+			return false, nil, err
+		}
+	}
+	secretData, ok := ecs.Data[common.EtcdEncryptionSecretFileName]
+	if !ok {
+		return true, nil, fmt.Errorf("EncryptionConfiguration in garden cluster (%v) does not contain expected element: %v", secretNameInGardenCluster, common.EtcdEncryptionSecretFileName)
+	}
+	ec, err := encryptionconfiguration.CreateFromYAML(secretData)
+	if err != nil {
+		return true, nil, fmt.Errorf("EncryptionConfiguration in garden cluster (%v) is not consistent: %v", secretNameInGardenCluster, err)
+	}
+	return true, ec, nil
+}
+
+// writeEncryptionConfiguration writes the secret which contains the EncryptionConfiguration to the
+// shoot namespace in the seed cluster as well as to the garden cluster
+func (b *Botanist) writeEncryptionConfiguration(ecYamlBytes []byte) error {
+	err := b.writeEncryptionConfigurationSecretToSeed(ecYamlBytes)
+	if err != nil {
+		return err
+	}
+	err = b.writeEncryptionConfigurationSecretToGarden(ecYamlBytes)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// writeEncryptionConfigurationSecretToSeed writes the secret which contains the EncryptionConfiguration
+// to the shoot namespace in the seed cluster
+func (b *Botanist) writeEncryptionConfigurationSecretToSeed(ecYamlBytes []byte) error {
+	b.Logger.Info("starting writeEncryptionConfigurationSecretToSeed")
+
+	client := b.Operation.K8sSeedClient
+	secretObj := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      common.EtcdEncryptionSecretName,
+			Namespace: b.Operation.Shoot.SeedNamespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			common.EtcdEncryptionSecretFileName: ecYamlBytes,
+		},
+	}
+	if _, err := client.CreateSecretObject(secretObj, true); err != nil {
+		return err
+	}
+	b.Logger.Info("EncryptionConfiguration written to seed successfully")
+	return nil
+}
+
+// writeEncryptionConfigurationSecretToGarden writes the secret which contains the EncryptionConfiguration
+// to the garden cluster
+func (b *Botanist) writeEncryptionConfigurationSecretToGarden(ecYamlBytes []byte) error {
+	b.Logger.Info("starting writeEncryptionConfigurationSecretToGarden")
+	client := b.Operation.K8sGardenClient
+	secretNameInGardenCluster := b.calculateEtcdEncryptionSecretNameInGardenCluster()
+	secretObj := &corev1.Secret{
```

Comment:
Except for the `ownerReferences` the `Secret` objects looks very similar to the one that is created in the seed. Maybe you can move the construction of this object into a separate function that takes `name`, `namespace`, and `ecYamlBytes` which returns the object. After that, you can add the `ownerReferences` here.

Reply:
done